trunner/testcase.py

```diff
@@ -107,6 +112,11 @@ def exec_test(self, proc):
                 self.exception = Color.colorify(msg, Color.BOLD)
                 self.handle_pyexpect_error(proc, exc)
                 return
+            except UnicodeDecodeError:
+                msg = 'Unicode Decode Error detected!\n'
+                self.exception = Color.colorify(msg, Color.BOLD)
```

Comment:
Can be moved into handle unicode error.  Additionaly passing original exception into handler would allow to print original message that contains information about problematic character.